src/stream/src/task/barrier_manager/managed_state.rs

```diff
@@ -287,16 +326,14 @@ impl ManagedBarrierState {
 
         match self.epoch_barrier_state_map.get_mut(&barrier.epoch.prev) {
             Some(&mut BarrierState {
-                curr_epoch,
-                inner:
-                    ManagedBarrierStateInner::Stashed {
-                        ref mut collected_actors,
-                    },
+                inner: ManagedBarrierStateInner::Stashed { .. },
                 ..
-            }) => {
-                let new = collected_actors.insert(actor_id);
-                assert!(new);
-                assert_eq!(curr_epoch, barrier.epoch.curr);
+            })
+            | None => {
+                panic!(
+                    "cannot collect new actor barrier {:?} at current state: Stashed or None",
+                    barrier.epoch,
+                )
```

Comment:
Am I understanding correctly: if the state is now `Stashed`, then it's impossible for an actor to call `collect` as the barrier message must be blocked at the `RemoteInput`.

Can we leave a comment here?

Reply:
Yes exactly. Let me add some comments here.